docs/how_to/retriever/root.md

```diff
@@ -0,0 +1,25 @@
+
+# 🔍 Retriever
+## Concept
+Retrievers specify how to look up the most relevant context from an [Index](/how_to/index/root.md)
```

Comment:
nit: maybe to be more precise, "Retrievers are responsible for fetching the most relevant context given a user query. It can be built on top of Indices, but can also be defined independently."

retrievers don't necc need to be defined on top of indices. relates to my comments below about linking the custom retriever tutorial

Reply:
good point.